featuremanagement/azuremonitor/_send_telemetry.py

```diff
@@ -51,17 +53,49 @@ def publish_telemetry(evaluation_event: EvaluationEvent) -> None:
     """
     if not HAS_AZURE_MONITOR_EVENTS_EXTENSION:
         return
-    event = {}
-    if evaluation_event.feature:
-        event[FEATURE_NAME] = evaluation_event.feature.name
+    event: Dict[str, Optional[str]] = {}
+    if not evaluation_event.feature:
+        return
+    event[FEATURE_NAME] = evaluation_event.feature.name
     event[ENABLED] = str(evaluation_event.enabled)
+    event["Version"] = EVALUATION_EVENT_VERSION
 
+    # VariantAllocationPercentage
     if evaluation_event.reason and evaluation_event.reason != VariantAssignmentReason.NONE:
         if evaluation_event.variant:
             event[VARIANT] = evaluation_event.variant.name
         event[REASON] = evaluation_event.reason.value
 
-    if evaluation_event.feature and evaluation_event.feature.telemetry:
+        if evaluation_event.reason == VariantAssignmentReason.DEFAULT_WHEN_ENABLED:
+            allocation_percentage = 0
+
+            if evaluation_event.feature.allocation and evaluation_event.feature.allocation.percentile:
+                for allocation in evaluation_event.feature.allocation.percentile:
+                    if (
+                        evaluation_event.variant
+                        and allocation.variant == evaluation_event.variant.name
+                        and allocation.percentile_to
```

Comment:
Why percentile_to is checked, but percentile_from is not checked?

Reply:
I suggest just remove this check since the allocation must be fine, otherwise the feature manager should have failed.

Reply:
I thought I posted this somewhere else. For type reasons we have `from` as an `int` and `to` as an `optional[int]`. We need to verify to as it is optional.